Sources/SwiftFusion/Inference/PPCATrackingFactor.swift

```diff
@@ -0,0 +1,119 @@
+import PenguinStructures
+import TensorFlow
+
+/// A factor over a target's pose and appearance in an image.
+public struct PPCATrackingFactor: LinearizableFactor2 {
+  /// The first adjacent variable, the pose of the target in the image.
+  public typealias V0 = Pose2
```

Comment:
V0 and V1 are not very intuitive names :-) Can we call them Pose and Latent?

Reply:
These are explicitly specifying `LinearizableFactor2`'s associatedtype `V0` and `V1` so they have to be named like this. I have added a comment explaining that.

Reply:
Well, but `V0` and `V1` are not awesome associated type names either, though I'm not sure we can do better.  I guess `V` stands for `Variable`?  I think if we named them `Variable0`, etc., or `AdjacentVariable0`, or `Neighbor0` the code might be more self-documenting.  The only question is, would other code become difficult for clients?

Reply:
I think that `Variable0` etc sounds really good. No client code uses these anywhere as far as I know, so it shouldn't cause any changes except making the declaration easier to read.

(In theory you don't need to declare these at all because associated type inference figures them out automatically. But in practice the error messages when associated type inference fails are so incomprehensible that I think we should really encourage every factor implementation to explicitly write them out.)

Reply:
I'll update the names in a separate PR though.